mod_lookup_identity.c

```diff
@@ -75,6 +75,37 @@ typedef struct lookup_identity_config {
 
 module AP_MODULE_DECLARE_DATA lookup_identity_module;
 
+static char * get_value_from_query(request_rec * r, const char * name) {
+	char * key, * val, * query, * stage = NULL;
+
+	ap_log_rerror(APLOG_MARK, APLOG_INFO, 0, r, "get_value_from_query invoked [%s, %s]", r->args, name);
+
+	if (r->args != NULL) {
+		if ((query = apr_pstrdup(r->pool, r->args)) == NULL) {
+			stage = "apr_pstrdup failed";
+			goto fail;
+		}
+	}
+	/* http://docstore.mik.ua/orelly/apache_mod/155.htm#BIN325 */
+	while(query && *query && (val = ap_getword(r->pool, (const char **)&query, '&'))) {
+		key = ap_getword(r->pool, (const char **)&val, '=');
+		ap_unescape_url(key);
+		ap_unescape_url(val);
+
+		if (strcmp(key, name) == 0) {
+			ap_log_rerror(APLOG_MARK, APLOG_INFO, 0, r, "get_value_from_query: return %s", val);
+			return apr_pstrdup(r->pool, val);
+		}
+	}
+
+	ap_log_rerror(APLOG_MARK, APLOG_INFO, 0, r, "get_value_from_query: %s not found", name);
+	return NULL;
+fail:
+	ap_log_rerror(APLOG_MARK, APLOG_ERR, 0, r, "get_value_from_query failed [%s]", stage);
+	return NULL;
+}
```

Comment:
Thank you, I was searching for this and was really wondering why something like this doesn't exits. Obviously I used wrong key words when searching.